### ISSUE TITLE
Using FreeDV with QO-100 satellite - enhancement request

### ISSUE THREAD

**Issue by barjac** [drowe67/freedv-gui#718]
I have been asked by Matt G6WPJ to forward this as an enhancement request to facilitate easier use of FreeDV in his rather complex satellite system.
(Matt is an RSGB news reader on GB2RS and broadcasts in FreeDV 700E every Sunday on 80m)
//-----------------
The following is an explanation in his own words:

I wonder if I may make a feature request for the next version of FreeDV. 

It is to help with using my TS-2000 and FreeDV on QO-100 although it will be true for anyone using a TS-2000 on any of its bands (HF/VHF/UHF).

The feature request concerns the issue we spoke about some weeks ago where in order to get the TS-2000 to use its rear panel audio port for transmit audio it is required to use the external key line present on the same rear panel connector to key the transmitter. Sending a CAT command via Hamlib causes the TS-2000 to use its front panel (Mic) audio input as the source for transmitted audio. 

The problem!
    In order to report my frequency of operation correctly on the FreeDV reporter I need to set the 'CAT' controls using 'hamlib' for a 'virtual' TS-2000 which is emulated by the SDR receiver software SDR console and appears on a virtual COM port (COM11) to external software such as FreeDV. 
    Once setup like this, SDR Console reports my operating frequency to FreeDV as 10,489.xxx MHz (e.g. 10,489.640 MHz) ; this is successfully passed on to FreeDV reporter and shows people I am on the QO-100 satellite. 
    I am only using the SDR Console software as a receiver; it has no transmit function in this case. 
    For transmit I am using a real TS-2000 operating on 144 MHz then upmixed to 2.4GHz for the QO-100 uplink. This is on a real COM port (COM3) but I cannot get FreeDV to use this as a key as the real TS-2000 and the virtual TS-2000 need to sync together so my receiver downlink frequency tracks with my transmitter uplink frequency. FreeDV cannot access the real TS-2000 serial port as it is already in use by the SDR Console software for the Tx/Rx frequency syncing function. 
    There is more detail to this but long and short is....I want FreeDV to understand that my transmitter is being keyed by a VOX function and act accordingly. 

The request!
    On the 'CAT and PTT Config' panel under the 'Hamlib Settings' entry boxes add a further option in the 'PTT uses:' dropdown menu. 
    The menu would then read CAT \ RTS \ DTR \ VOX \ None (Rx Only)
    Selecting VOX would not do anything in the software as such. 
        If I assume the 'half-duplex' option is selected 
        With the VOX option selected, pressing the PTT button in the main GUI would turn on the Tx audio as normal and send a PTT active status to FreeDV reporter to turn my display line red as normal. 

In other words selecting VOX means nothing to actually do from the FreeDV software to key the radio (no CAT message or asserting RTS or DTR) just implicitly know that if the Tx audio is on; the radio is keyed and tell the FreeDV reporter accordingly. 

This is different to 'None' which means I do not have a transmitter or the other options which tell the software to do something active to 'key' the Tx.

I hope I have explained this clearly but if you could feed this request in I would be very grateful.

**Comment by tmiw** [comment]
The behavior of "None" actually changed for 1.9.9.1 in that it no longer disables the PTT button (and no longer causes you to show up as "RX Only" in FreeDV Reporter). This was done to fix a bug involving PTT input (https://github.com/drowe67/freedv-gui/pull/707). Sounds like that change might be sufficient for this use case?

**Comment by barjac** [comment]
Thanks Mooneer,
I alerted Matt to your comment but he is very busy this week.
I will report back as and when I hear from him.

**Comment by barjac** [comment]
OK Matt confirms that updating to 1.9.9.1 in the system in question has indeed fixed this issue.
He passes on his thanks for your prompt reply as always :)

So closing.
